# Midterm1 Data Preview

This notebook will allow you to explore the data you will use on the midterm exam on Wednesday 27 September.

I encourage you to imagine the kinds of questions I might ask you so that you can practice to be ready for the exam.

The dataset has **5 columns and many rows**. 

Each row represents the number of books read on Tar Heel Reader on a day. 

The first day is a Sunday and no days are missing.

You may **not** assume the number of days will be the same when I grade the exam but you **may** assume the number of days will be a multiple of 7.

You **may** assume five columns.

The first column gives the number of books read on that day all over the world. The second through fifth columms are the number of books read in the top four users of the site: United States, Canada, Australia, and New Zealand. 

Let's load the data and look at a few rows.

In [2]:
# don't change any of this
%matplotlib inline
import numpy as np
import pylab

# No need to redo this. Just use the data array for everything
data = np.loadtxt('daily2016.txt')
data.flags.writeable = False # trying to protect you from yourself

# look at the first 5 rows
data[:5,:]

array([[ 738.,  649.,    5.,   22.,    0.],
       [ 603.,  493.,   15.,    1.,    0.],
       [ 814.,  578.,   64.,   35.,    9.],
       [ 822.,  679.,    7.,   27.,    0.],
       [ 336.,  259.,   27.,   40.,    0.]])

From the above, I can tell that for this dataset on the first Sunday:

* 738 books were read all over the world
* 649 of those were read in the US
* 5 in Canada
* 22 in Australia
* and none in New Zealand



Now, I encourage you to ask and answer questions about these data. 

You might ask questions like:

* How many days of data do we have?
* What were the total number of books read? (Watch out, its not the sum of all of the numbers)
* And many more...

In [3]:
np.count_nonzero(data, axis=0)

array([371, 371, 370, 371, 341])